pkg/controllers/disruption/orchestration/queue.go

```diff
@@ -305,53 +309,57 @@ func (q *Queue) waitOrTerminate(ctx context.Context, cmd *Command) error {
 // Add adds commands to the Queue
 // Each command added to the queue should already be validated and ready for execution.
 func (q *Queue) Add(cmd *Command) error {
-	providerIDs := lo.Map(cmd.candidates, func(s *state.StateNode, _ int) string {
-		return s.ProviderID()
-	})
 	// First check if we can add the command.
-	if q.HasAny(providerIDs...) {
+	if q.HasAny(cmd.candidates...) {
 		return fmt.Errorf("candidate is being disrupted")
 	}
 
+	q.Lock()
+	defer q.Unlock()
 	cmd.timeAdded = q.clock.Now()
-	q.mu.Lock()
-	for _, candidate := range cmd.candidates {
-		q.providerIDToCommand[candidate.ProviderID()] = cmd
+	for _, c := range cmd.candidates {
+		q.nodeClaimsInQueueSet.Insert(c.NodeClaim.Status.ProviderID)
 	}
-	q.mu.Unlock()
-	q.TypedRateLimitingInterface.Add(cmd)
+	q.providerIDToCommand[cmd.KeyNodeClaim.Status.ProviderID] = cmd
+	q.source <- event.TypedGenericEvent[*v1.NodeClaim]{Object: cmd.KeyNodeClaim}
 	return nil
 }
 
 // HasAny checks to see if the candidate is part of an currently executing command.
-func (q *Queue) HasAny(ids ...string) bool {
-	q.mu.RLock()
-	defer q.mu.RUnlock()
+func (q *Queue) HasAny(candidates ...*state.StateNode) bool {
+	q.RLock()
+	defer q.RUnlock()
 
 	// If the mapping has at least one of the candidates' providerIDs, return true.
-	_, ok := lo.Find(ids, func(id string) bool {
-		_, ok := q.providerIDToCommand[id]
-		return ok
+	_, ok := lo.Find(candidates, func(n *state.StateNode) bool {
+		return q.nodeClaimsInQueueSet.Has(n.ProviderID())
 	})
 	return ok
 }
 
+// Used for testing, gets the commands in the current queue
+func (q *Queue) GetCommands() []*Command {
```

Comment:
To me: This is somewhat weird to me -- we don't have similar testing that tests what controller-runtime has in its queue and it feels like if we just keep our functional testing, we reduce the amount of changes that we need to make here and keep our testing more functional

Reply:
Discussed offline: This is fine to me so we don't take a dependency on the thing that we use as the "key" in the orchestration queue